src/components/Datepicker/Datepicker.js

```diff
@@ -57,19 +80,20 @@ export class Datepicker extends React.Component {
 
     let initial;
     let isValid;
+    const isControlled = Datepicker.isControlled(date);
 
-    if (date) {
+    if (isControlled) {
       initial = date;
       isValid = Datepicker.validate(date);
-    } else if (!date && initialDate) {
+    } else {
       initial = initialDate;
       isValid = Datepicker.validate(initialDate);
     }
 
     this.state = {
       open,
       inputValue: isValid ? moment(initial).format(defaultDateFormat).toString() : '',
-      isInvalid: !isValid,
+      isValid: true,
```

Comment:
because initially we don't want to see an error message. but as soon as the user clicks on a required input field or enters a wrong date, the error message gets displayed.